src/app/cli/CommandLineInterface.java

```diff
@@ -63,15 +92,280 @@ public void launch(){
-                    System.out.println(user);
+                    this.user = app.logIn(tokens[1], tokens[2]);
+                    System.out.println(this.user);
                 }
                 case LOGOUT -> {
                     app.logOut();
                 }
                 case SIGNUP -> {
-                    User user = app.signUp(tokens[1], tokens[2], tokens[3], tokens[4], tokens[5]);
-                    System.out.println(user);
+                    this.user = app.signUp(tokens[1], tokens[2], tokens[3], tokens[4], tokens[5]);
+                    System.out.println(this.user);
+                }
+
+                case GET_COLLECTIONS -> {
+                    Collection[] collections = app.get_collections();
+                    if(collections.length == 0) {
+                        System.out.println("You do not have any collections.");
+                        continue;
+                    }
+                    for(int i = 0; i < collections.length; ++i){
+                        for(int j = i+1; j < collections.length; ++j){
+                            Collection first = collections[i];
+                            Collection second = collections[j];
+                            if(first.coll_name().compareTo(second.coll_name()) > 0) {
+                                collections[i] = second;
+                                collections[j] = first;
+                            }
+                        }
+                    }
+                    for (Collection collection : collections) {
+                        System.out.println("Name:\t" + collection.coll_name());
+                        System.out.println("# of games\t" + collection.games().length);
+                        System.out.println("Play time:\t" + app.total_playtime_collection(collection));
+                    }
+                    System.out.println();
+                }
+
+                case COLLECTION_ADD -> {
+                    Collection[] collections_list_init = app.get_collections();
+                    ArrayList<Collection> collections_list = new ArrayList<>();
+                    for (Collection collection : collections_list_init) {
+                        if (collection.coll_name().equals(tokens[1])) {
+                            collections_list.add(collection);
+                        }
+                    }
+                    if(collections_list.size() == 0) {
+                        System.out.println("You do not have a collection with that name.");
+                        continue;
+                    }
+                    Collection selected_coll = collections_list.get(0);
+
+                    if(collections_list.size() > 1) {
+                        System.out.println("Which collection would you like to add the game to (enter the number)?");
+                        for(int i = 0; i < collections_list.size(); ++i) {
+                            Collection curr_coll = collections_list.get(i);
```

Comment:
printing a `Colleciton` should probably be put in its own method

Reply:
I do agree with this, will implement